### PR TITLE
MGMT-8943: Update documentation to V2 REST-API

### PR DESCRIPTION
**Pull request by nmagnezi**
This change update the current documentation to use V2 REST-API endpoints.
It also addes some syntax highlighting and other corrections.
 with '#' will be ignored, and an empty message aborts the commit.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @filanov 
/cc @avishayt 
/cc @romfreiman 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/user-guide/install-customization.md [hunk 1]
```diff
@@ -50,13 +50,13 @@ curl \
     --header "Authorization: Bearer $TOKEN" \
     --request PATCH \
     --data @discovery-ign.json \
-"http://$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install/v1/clusters/$CLUSTER_ID/discovery-ignition"
+"http://$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install//v2/infra-envs/{infra_env_id}"
```
Comment (avishayt) [inline docs/user-guide/install-customization.md:53]:
```suggestion
"http://$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install/v2/infra-envs/{infra_env_id}"
```

**Review by avishayt — COMMENTED** [review]
Looks great, just a minor change

File: docs/user-guide/install-customization.md (hunk 1 above)
Comment (nmagnezi) [reply docs/user-guide/install-customization.md]: Done

File: docs/set-discovery-password.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ echo ============================ >> .set_iso_password_log
 echo >> .set_iso_password_log
 
 echo Telling service to use our patched ignition file
-curl --fail -s $OCM_API_ENDPOINT/assisted-install/v1/clusters/$CLUSTER_ID/discovery-ignition -H "Authorization: Bearer $TOKEN" --request PATCH --header "Content-Type: application/json" --data @<(echo '{"config": "replaceme"}' | jq --rawfile ignition <(echo $NEW_IGNITION) '.config = $ignition')
+curl --fail -s $OCM_API_ENDPOINT/assisted-install/v2/infra-envs/{infra_env_id} -H "Authorization: Bearer $TOKEN" --request PATCH --header "Content-Type: application/json" --data @<(echo '{"ignition_config_override": "replaceme"}' | jq --rawfile ignition <(echo $NEW_IGNITION) '.config = $ignition')
```
Comment (filanov) [inline docs/set-discovery-password.md:78]:
```suggestion
curl --fail -s $OCM_API_ENDPOINT/assisted-install/v2/infra-envs/$INFRA_ENV_ID -H "Authorization: Bearer $TOKEN" --request PATCH --header "Content-Type: application/json" --data @<(echo '{"ignition_config_override": "replaceme"}' | jq --rawfile ignition <(echo $NEW_IGNITION) '.config = $ignition')
```

File: docs/set-minimal-iso.md [hunk 3]
```diff
@@ -46,9 +46,9 @@ function log() {
 trap log EXIT
 
 echo Telling service to generate a minimal ISO with our public SSH key file
-curl -fail -s $OCM_API_ENDPOINT/assisted-install/v1/clusters/$CLUSTER_ID/downloads/image -H "Authorization: Bearer $TOKEN" --request POST --header "Content-Type: application/json" --data @<(echo '{"image_type": "minimal-iso", "ssh_public_key": ""}' | jq --rawfile pubkey <(echo -n $SSH_KEY) '.ssh_public_key = $pubkey')
+curl -fail -s $OCM_API_ENDPOINT/assisted-install/v2/infra-envs/$INFRA_ENV_ID -H "Authorization: Bearer $TOKEN" --request POST --header "Content-Type: application/json" --data @<(echo '{"image_type": "minimal-iso", "ssh_authorized_key": ""}' | jq --rawfile pubkey <(echo -n $SSH_KEY) '.ssh_authorized_key = $pubkey')
```
Comment (filanov) [inline docs/set-minimal-iso.md:49]: shouldn't line 29 change as well? (i can't comment it)
```
if [ -z ${INFRA_ENV_ID+x} ]; then
	echo 'Please set INFRA_ENV_ID to your infra-env ID, which can be found in the Assisted Installer URL'
	exit 1
fi
```

File: docs/user-guide/install-customization.md (hunk 1 above)
Comment (filanov) [inline docs/user-guide/install-customization.md:53]: Just to keep the same flow in the file. 
```suggestion
"http://$ASSISTED_SERVICE_IP:$ASSISTED_SERVICE_PORT/api/assisted-install//v2/infra-envs/$INFRA_ENV_ID"
```

File: docs/set-minimal-iso.md (hunk 3 above)
Comment (nmagnezi) [reply docs/set-minimal-iso.md:49]: Done

**Review by filanov — APPROVED** [review]

**Review by nmagnezi — COMMENTED** [review]
Addressed all comments. Done.

**Pull request merged**